data-prepper-core/src/main/java/com/amazon/dataprepper/plugin/PluginPackagesSupplier.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugin;
+
+import com.google.common.collect.Iterators;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * Supplies packages to scan for plugins.
+ */
+class PluginPackagesSupplier implements Supplier<String[]> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PluginPackagesSupplier.class);
+    private static final String DEFAULT_PLUGINS_CLASSPATH = "com.amazon.dataprepper.plugins";
+
+    PluginPackagesSupplier() {
+    }
+
+    Iterator<URL> loadResources() throws IOException {
+        final Enumeration<URL> resourcesEnumeration = getClass().getClassLoader().getResources("META-INF/data-prepper.plugins.properties");
+        return Iterators.forEnumeration(resourcesEnumeration);
+    }
+
+    @Override
+    public String[] get() {
+
+        final Set<String> packageNames = getUniquePackageNames();
+
+        final String[] packageNamesArray = packageNames.toArray(new String[0]);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Using packages for plugins: {}", String.join(",", packageNamesArray));
+        }
+
+        return packageNamesArray;
+    }
+
+    private Set<String> getUniquePackageNames() {
+        Set<String> packageNames;
+        try {
+            final Iterator<URL> pluginResources = loadResources();
+            packageNames = readResources(pluginResources);
+        } catch (final IOException ex) {
+            LOG.warn("Unable to load data-prepper.plugins.properties file. Reverting to default plugin package.");
+            packageNames = Collections.singleton(DEFAULT_PLUGINS_CLASSPATH);
+        }
+        return packageNames;
+    }
+
+    private Set<String> readResources(final Iterator<URL> pluginResources) {
+        final Set<String> packageNames = new HashSet<>();
+        while (pluginResources.hasNext()) {
+
+            final URL pluginPropertiesUrl = pluginResources.next();
+
+            final Properties pluginProperties = new Properties();
+            final InputStream pluginPropertiesInputStream;
+            try {
+                pluginPropertiesInputStream = pluginPropertiesUrl.openStream();
+                pluginProperties.load(pluginPropertiesInputStream);
+            } catch (final IOException ex) {
+                LOG.warn("Unable to load properties from resource. url={}", pluginPropertiesUrl, ex);
+            }
+
+            final String packagesRawString = pluginProperties.getProperty("org.opensearch.dataprepper.plugin.packages");
+
+            final String[] currentPackageNames = packagesRawString.split(",");
```

Comment:
`getProperty` will return null if "org.opensearch.dataprepper.plugin.packages" does not exist. That would result in a NPE on line 83.